widgets/matplotlib/measurement/depth_profile.py

```diff
@@ -298,7 +307,23 @@ def _make_legend_box(self) -> None:
         leg = self.axes.legend(
             handles, labels_w_percentages, loc=3, bbox_to_anchor=(1, 0),
             borderaxespad=0, prop={"size": 11, "family": "monospace"})
-
+        
+        # If "Show used efficiency files" is checked and text-object is not
+        # yet created.
+        if self._show_eff_files and self.eff_text is None:
+            eff_str = self.used_eff_str.replace("\t","") 
+            
+            # Set position of text according to amount of lines in the string
+            line_count = eff_str.count("\n") + 1
+            yposition_txt = 1 - 0.08 * line_count
```

Comment:
I think there could be a better way to adjust the vertical position of the text.